### ISSUE TITLE
dog2cat is not as good

### ISSUE THREAD

**Issue by zhihongp** [pix2pixzero/pix2pix-zero#11]
There is a dog2cat task in the code but when tested using AFHQ, the result is pretty bad (some end up with black images) comparing to published cat2dog results.  Is there necessary steps missed for the new dog2cat task, like it needs fine-tune of certain parameters for each task?

**Comment by zhihongp** [comment]
Just ran the cat2dogtask too, the black image is also present (pixabay_cat_000165.png) , just not as much as dog2cat.  Am I missing some settings here.  I am using `--negative_guidance_scale 5`

**Comment by pix2pixzero** [comment]
Hi, 

The black image represents that some NSFW content was detected in the output.
This can sometimes happen even if the image does not contain any NSFW content, as the NSFW detector is not perfect. 
You could perhaps try disabling the filter. 

- Authors

**Comment by zhihongp** [comment]
Thanks for the tip.  After turning off the filter, no more black output but those are still failure cases, like pixabay_dog_001059.png is translated to a blurry object of skin tone.  Is this kind of outliers expected?

**Comment by pix2pixzero** [comment]
This is not expected. Could you attach the file and the parameters you use? 
I can try this out and see why the results are not as good. 

-Authors

**Comment by zhihongp** [comment]
This is one example that triggered NSFW check.

Also I have noticed the edited image is randomly generated each time.  Is there a way to fix the random seed so results could be duplicated?

> env CUDA_VISIBLE_DEVICES=$1 python src/inversion.py  \
    --input_image "assets/pixabay_cat_001309.png" \
    --results_folder "output/test" \
    --num_ddim_steps 50
env CUDA_VISIBLE_DEVICES=$1 python src/edit_real.py \
    --inversion "output/test/inversion/pixabay_cat_001309.pt" \
    --prompt "output/test/prompt/pixabay_cat_001309.txt" \
    --task_name "cat2dog" \
    --results_folder "output/test" \
    --num_ddim_steps 50 \
    --negative_guidance_scale 5

**Comment by sayakpaul** [comment]
If you're conditioning the pipeline based on an input image then `torch.manual_seed(your_seed)` is the easiest option. 

But if you're generating based on an input prompt, then you might want to fix the initial latent. 

```
import torch

torch.manual_seed(0)

initial_latent = torch.randn(1, 4, 64, 64) # Change the batch size accordingly
```

Now while calling the pipeline, set `x_in` to `initial_latent`. 

The first case that I mentioned is implemented in all the test cases we implement in `diffusers. See [here](https://github.com/huggingface/diffusers/blob/main/tests/pipelines/stable_diffusion/test_stable_diffusion_pix2pix_zero.py) for an example.
